src/digest/sha1.rs

```diff
@@ -49,38 +49,46 @@ fn block_data_order_safe(state: &mut [u64; MAX_CHAINING_LEN / 8],
     let state = &mut state[..CHAINING_WORDS];
     let state = slice_as_array_ref_mut!(state, CHAINING_WORDS).unwrap();
 
-    let mut w: [W32; 80] = [Wrapping(0); 80];
+    let mut w: [W32; 16] = [Wrapping(0); 16];
     for block in blocks {
-        for t in 0..16 {
-            let word = slice_as_array_ref!(&block[t * 4..][..4], 4).unwrap();
-            w[t] = Wrapping(polyfill::slice::u32_from_be_u8(word))
-        }
-        for t in 16..80 {
-            let wt = w[t - 3] ^ w[t - 8] ^ w[t - 14] ^ w[t - 16];
-            w[t] = polyfill::wrapping_rotate_left_u32(wt, 1);
-        }
-
         let mut a = state[0];
         let mut b = state[1];
         let mut c = state[2];
         let mut d = state[3];
         let mut e = state[4];
 
-        for t in 0..80 {
+        macro_rules! round {
```

Comment:
Does this really need to be a macro? It seems like it could be an inline function, albeit one that takes a lot of arguments. I'd rather avoid macros unless they are necessary for some purpose, in which case we should document why we use a macro instead of a function.